docs/GetStarted/getting_started_macos_bazel.md

```diff
@@ -0,0 +1,164 @@
+# Getting Started on MacOS with Bazel
+
+<!--
+Notes to those updating this guide:
+
+    * This document should be __simple__ and cover essential items only.
+      Notes for optional components should go in separate files.
+
+    * This document parallels getting_started_linux_bazel.md and
+      getting_started_windows_bazel.md
+      Please keep them in sync.
+-->
+
+This guide walks through building the core compiler and runtime parts of IREE
+from source. Auxiliary components like the Python bindings and Vulkan driver are
+not documented for MacOS at this time.
+
+IREE is not officially supported on MacOS at this time. It may work, but it is
+not a part of our open source CI, and may be intermittently broken.
+Contributions related to MacOS support and documentation are welcome however.
+
+## Prerequisites
+
+### Install Bazel
+
+Install the version of Bazel found in
+[`.bazelversion`](https://github.com/google/iree/blob/master/.bazelversion).
+
+#### Install xcode.
+
+If you've already installed `xcode-select` and this fails with it can likely be
+ignored.
+
+```
+$ sudo xcodebuild -license accept
+```
+
+#### Download and install Bazel from a binary.
+```
+# Example using Bazel version `2.1.0`
+$ export BAZEL_VERSION=2.1.0
+
+# Download and install
+$ curl -LO "https://github.com/bazelbuild/bazel/releases/download/${BAZEL_VERSION}/bazel-${BAZEL_VERSION}-installer-darwin-x86_64.sh"
+$ chmod +x "bazel-${BAZEL_VERSION}-installer-darwin-x86_64.sh"
+$ ./bazel-${BAZEL_VERSION}-installer-darwin-x86_64.sh
+
+# Add to path
+$ echo -e '\n# Add bazel to PATH:\nexport PATH="${PATH}:${HOME}/bin"' >> ~/.zshrc
+$ source ~/.zshrc
+
+# Confirm the version
+$ bazel --version # 2.1.0
+```
+
+See the
+[official docs](https://docs.bazel.build/versions/master/install-os-x.html#install-with-installer-mac-os-x)
+for additional guidance if needed.
```

Comment:
Is there anything here that differs from the official docs? We can just link to them here instead of including all these details in our own docs.

Reply:
There were a few things:
- Following their documentation for installing Xcode led to errors on my system and it was difficult to tell what to do until I decided to just try moving on. In the end it looks like installing `xcode-select` is sufficient here, and takes up a lot less space than the entire `Xcode.app`.
- Their documentation isn't copy/paste-able, so adding an env variable for the version made it easier for me to follow.
- They recommend adding the path to `.bashrc`, which isn't useful because MacOS uses ZSH by default. I added a command that appends the relevant command to their `.zshrc`.

We could turn this section into the example below to match the other files, people will just may have to debug things.

Install Bazel version > 2.0.0 (see [`.bazelversion`](https://github.com/google/iree/blob/master/.bazelversion) for the specific version IREE uses) by following the [official docs](https://docs.bazel.build/versions/master/install-os-x.html#install-with-installer-mac-os-x).

Reply:
I'd still prefer to just point people at the official docs for any dependencies. If there are issues they should be fixed in those docs, not in projects like ours. At the same time, we should assume that developers are able to set up their own environments as they choose (e.g. using different package managers, installing to different locations). If people have further difficulties they can file issues or get in contact with us in other ways.

Reply:
Changed to mirror the Linux Bazel docs!